airflow/providers/amazon/aws/transfers/snowflake_to_s3.py

```diff
@@ -0,0 +1,166 @@
+        'snowflake' (default) to use the internal Snowflake authenticator
+        'externalbrowser' to authenticate using your web browser and
+        Okta, ADFS or any other SAML 2.0-compliant identify provider
+        (IdP) that has been defined for your account
+        'https://<your_okta_account_name>.okta.com' to authenticate
+        through native Okta.
+    :type authenticator: str
+    :param session_parameters: You can set session-level parameters at
+        the time you connect to Snowflake
+    :type session_parameters: dict
+    """
+
+    template_fields = (
+        's3_path',
+        's3_bucket',
+        'sql',
+        'snowflake_conn_id',
+    )
+    template_ext = ('.sql',)
+    template_fields_renderers = {"sql": "sql"}
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        s3_path: str,
+        snowflake_conn_id: str = "snowflake_default",
+        schema: Optional[str] = None,
+        table: Optional[str] = None,
+        sql: Optional[str] = None,
+        stage: Optional[str] = None,
+        s3_bucket: Optional[str] = None,
+        file_format: Optional[str] = None,
+        unload_options: Optional[list] = None,
+        include_header: Optional[bool] = False,
+        database: Optional[str] = None,
+        warehouse: Optional[str] = None,
+        role: Optional[str] = None,
+        authenticator: Optional[str] = None,
+        session_parameters: Optional[dict] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.s3_path = s3_path
+        self.snowflake_conn_id = snowflake_conn_id
+        self.schema = schema
+        self.table = table
+        self.sql = sql
+        self.stage = stage
+        self.s3_bucket = s3_bucket
+        self.file_format = file_format
+        self.unload_options: List = unload_options or []
+        self.include_header = include_header
+        self.database = database
+        self.warehouse = warehouse
+        self.role = role
+        self.authenticator = authenticator
+        self.session_parameters = session_parameters
+
+    def execute(self, context):
+        snowflake_hook = SnowflakeHook(
+            snowflake_conn_id=self.snowflake_conn_id,
+            warehouse=self.warehouse,
+            database=self.database,
+            role=self.role,
+            authenticator=self.authenticator,
+            session_parameters=self.session_parameters,
+        )
+        unload_options = '\n\t\t\t'.join(self.unload_options)
+
+        if self.sql:
+            self.sql = f"({self.sql})"
+        else:
+            self.sql = self.schema + '.' + self.table
+
+        if self.stage:
+            unload_query = f"""
+            COPY INTO @{self.stage}/{self.s3_path}
+            FROM {self.sql}
+            """
+        else:
+            unload_query = f"""
+            COPY INTO 's3://{self.s3_bucket}/{self.s3_path}'
+            FROM {self.sql}
+            STORAGE_INTEGRATION = S3
```

Comment:
It should be customizable. Each customer can create one or more integration with different names and ARN roles See: https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration.html